Initial release of Illusion of Labyrinth (#8506)
rathena/rathena: npc/re/quests/quests_illusion_dungeons.txt

	npctalk "Teddy Bear: What have you done to me...", "", bc_self;
	end;
}

//============================================================
//= Illusion of Labyrinth
//============================================================
prt_maze01,99,27,0	script	#iol000	HIDDEN_WARP_NPC,4,4,{
	end;
OnTouch:

> 	if (BaseLevel < 180)

Reviewer: ```suggestion
	if (BaseLevel < 170)
```

		close;
	case 1:
		mes "[Anes]";
		mes "There are still uncompleted quests. ^ff0000If you accept new quests now, the unfinished quests from last week will be deleted.^000000 Would you like to accept new weekly quests?";
		next;
		if (select( "Cancel", "Receive the weekly request" ) == 1) {
			mes "[Anes]";
			mes "Come back when you're ready to accept the quest.";
			close;
		}
		if (isbegin_quest(12478) > 0)
			erasequest 12478;
		if (isbegin_quest(12479) > 0)
			erasequest 12479;
		if (isbegin_quest(12480) > 0)
			erasequest 12480;
		if (isbegin_quest(12481) > 0)
			erasequest 12481;
		setquest 12491;
		setquest rand(12478, 12481);
		mes "[Anes]";
		mes "May God bless you this week as well.";
		close;
	case 2:
		end;
	}
	end;

OnInit:
	setunittitle( getnpcid(0), "<Priest Guild New Member>" );

	questinfo( QTYPE_EVENT, QMARK_YELLOW, "ill_laby == 6 && checkquest(12491,PLAYTIME) == -1" );	// note: type event is not a mistake

	// no questinfo when playtime 2
	// questinfo( QTYPE_EVENT, QMARK_YELLOW, "checkquest(12491,PLAYTIME) == 2" );

	// unknown questinfo hunting completed
	end;
}

prt_maze01,99,23,5	script(CLOAKED)	Twisted Crack#1	CLEAR_NPC,{

> 	if (BaseLevel < 180)

Reviewer: ```suggestion
	if (BaseLevel < 170)
```

		end;
	if (ill_laby < 2) {
		mes "Beyond the twisted crack, a familiar yet unfamiliar labyrinth unfolds.";
		close;
	}
	mes "Beyond the twisted crack, a familiar yet unfamiliar labyrinth unfolds.";
	mes "^ff0000If you step in, you may not be able to get out again.^000000";
	next;
	if (select( "Pretend you didn't see it", "Enter through the crack" ) == 1) {
		mes "^0000ffCuriosity killed the cat. Let's bury it quietly.^000000";
		close;
	}
	warp "prt_mz03_i",99,26;
	end;

OnInit:
	questinfo( QTYPE_CLICKME, QMARK_YELLOW, "isbegin_quest(12484) == 1 || isbegin_quest(12485) == 1 || isbegin_quest(12486) == 1 || isbegin_quest(12487) == 1" );
	end;
}

prt_maze01,97,26,5	script(CLOAKED)	Irene#1	4_F_ERENE,2,2,{

> 	if (BaseLevel < 180)

Reviewer: ```suggestion
	if (BaseLevel < 170)
```

			mes "[Irene]";
			mes "Well yeah.. it must have been an unreasonable request... sorry..";
			close;
		}
		mes "[Irene]";
		mes "Thank you! Here's a note with the guild leader's location! Please!";
		next;
		select( "Let's go outside first." );
		ill_laby = 1;
		setquest 12482;
		warp "prt_fild01",136,365;
		end;
	}
	if (ill_laby == 1) {
		mes "[Irene]";
		mes "If you look at the note I gave you, the guild leader's location is written down.";
		close2;
		warp "prt_fild01",136,365;
		end;
	}
	if (ill_laby == 2) {
		mes "[Irene]";
		mes "Ugh, it looks like there's no trace of my remains...";
		next;
		mes "[Esmeralda]";
		mes "Fortunately, your guardian relic is stored in the guild, so resurrection is possible. It might cost some money, though.";
		next;
		mes "[Irene]";
		mes "I'm sorry...";
		close;
	}
	if (ill_laby == 3) {
		mes "[Irene]";
		mes "I need to go back to the guild soon. It's quite embarrassing because of the state of my soul.";
		close2;
		cloakonnpcself();
		end;
	}
	end;

OnTouch:
	if (ill_laby == 0) {
		npctalk "Ouch! Sorry. I didn't mean to surprise you!", "", bc_self;
		end;
	}
	end;

OnInit:
	setunittitle( getnpcid(0), "<Priest Guild Member>" );

> 	questinfo( QTYPE_QUEST, QMARK_YELLOW, "BaseLevel >= 180 && ill_laby == 0" );

Reviewer: ```suggestion
	questinfo( QTYPE_QUEST, QMARK_YELLOW, "BaseLevel >= 170 && ill_laby == 0" );
```

		mes "Irene, stop crying and come to your senses.";
		next;
		mes "[Esmeralda]";
		mes "Fortunately, your guardian relic is stored in the guild, so resurrection is possible. It might cost some money, though.";
		next;
		mes "[Irene]";
		mes "I'm sorry...";
		next;
		mes "[Esmeralda]";
		mes "We need to resurrect this troublemaker as soon as possible, so I'll send her back to the guild first.";
		next;
		mes "[Esmeralda]";
		mes "I have to say thank you, but since you have already helped me, can I ask you one more favor?";
		next;
		if (select( "You really have no shame.", "Uh... of course." ) == 1) {
			mes "[Esmeralda]";
			mes "Ah, I guess that's it. I really have no shame. We'll take care of the rest.";
			close;
		}
		mes "[Esmeralda]";
		mes "Irene made a soul contract with the guardian relic a long time ago, so I don't think there will be any problems with her resurrection.";
		next;
		mes "[Esmeralda]";
		mes "There are a total of 4 more people missing. These guys don't have a contract with the guardian relic, so they don't appear in spirit form. Ha... really...";
		next;
		select( "Is signing a contract with a guardian relic like insurance?" );
		mes "[Esmeralda]";

> 		mes "Yes, since our guild is made up of only priests, fatal accidents are unexpectedly rare, but they are very frequent.";

Reviewer 2: nexpectedly rare => very frequent.
Kind of contradicting?

		next;
		mes "[Esmeralda]";
		mes "As soon as you join, you enter into a contract with a guardian relic and undergo a ceremony that completely resurrects your soul and body even after death.";
		next;
		mes "[Esmeralda]";

> 		mes "The problem is that the four new recruits who disappeared today are the ones who didn't even go through that process.";

Reviewer 2: ```suggestion
		mes "The problem is, that the 4 new recruits, who disappeared today, didn't even go through that process.";
```
Was always as number so far.

		next;
		select( "Do I have to retrieve the remains and bring them back?" );
		mes "[Esmeralda]";
		mes "Ah... well, even the smallest traces of the body are fine. The more intact it is, the cheaper the resurrection price will be, but you have to save it first.";
		next;
		select( "(It feels like furniture repair)" );
		mes "[Esmeralda]";
		mes "Priests train in various things. During the training process, there are many things that are not quite beautiful.";
		next;
		if (select( "Well, it looks dangerous so I...", "Alright, I will do it" ) == 1) {
			mes "[Esmeralda]";
			mes "Ah, I guess that's it. We'll take care of the rest.";
			close;
		}
		mes "[Esmeralda]";

> 		mes "Thank you. I've collected information about the four guild members, so if you need anything, please check it. Perhaps traces of my members lie beyond this twisted rift.";

Reviewer 2: ```suggestion
		mes "Thank you. I've collected information about the 4 guild members, so if you need anything, please check it. Perhaps traces of my members lie beyond this twisted rift.";
```

		close2;
		ill_laby = 3;
		erasequest 12483;
		setquest 12484;
		setquest 12485;
		setquest 12486;
		setquest 12487;
		end;
	}
	if (ill_laby == 3) {
		mes "[Esmeralda]";

> 		mes "I've collected information about the four guild members, so if you need anything, please check it. Perhaps traces of my members lie beyond this twisted rift.";

Reviewer 2: ```suggestion
		mes "I've collected information about the 4 guild members, so if you need anything, please check it. Perhaps traces of my members lie beyond this twisted rift.";
```

		next;
		mes "[Esmeralda]";
		mes "Oh, and it's a bit difficult to show our guild's soul summoning process if you're not a member of the guild. Could you please go outside the dungeon for a moment?";
		next;
		if (select( "Can't I just watch?", "Of course" ) == 1) {
			mes "[Esmeralda]";
			mes "Ah, it would be a bit awkward to show this to someone outside the guild...";
			close;
		}
		mes "[Esmeralda]";
		mes "Thank you for your consideration. Then I'll send you outside for a while.";
		close2;
		ill_laby = 5;
		erasequest 12488;
		setquest 12489;
		warp "prt_fild01",136,365;
		end;
	}
	end;

OnInit:
	setunittitle( getnpcid(0), "<Priest Guild Leader>" );

	questinfo( QTYPE_QUEST, QMARK_YELLOW, "isbegin_quest(12483) == 1" );
	questinfo( QTYPE_QUEST, QMARK_YELLOW, "isbegin_quest(12488) == 1" );
	end;
}

prontera,212,320,0	script	#iol002	HIDDEN_WARP_NPC,4,4,{
	end;
OnTouch:
	if (ill_laby == 1) {
		cloakoffnpcself( "Esmeralda#2" );
		end;
	}
	end;
}

prontera,212,320,3	script(CLOAKED)	Esmeralda#2	4_GEFFEN_09,2,3,{
	if (ill_laby == 1) {
		mes "[Esmeralda]";
		mes "Huh? What's going on? Did something happen to our guild members?";
		next;

> 		select( "(Inform her terrible thing that happened to the priests' guild)" );

Reviewer 2: ```suggestion
		select( "(Inform her about the terrible thing that happened to the priest guild)" );
```

		mes "[Esmeralda]";
		mes "Our new recruits are missing?";

> 		mes "Sigh... Is it a guild tradition? It always like this every year...";

Reviewer 2: ```suggestion
		mes "Sigh... Is it a guild tradition? It is always like this every year...";
```

		next;
		mes "[Esmeralda]";
		mes "Thanks for telling me. I guess I'll have to go take care of these guys. God's blessings be with you, adventurer...";
		close2;
		ill_laby = 2;
		erasequest 12482;
		setquest 12483;
		cloakonnpcself();
		end;
	}
	end;

OnTouch:
	if (ill_laby == 1) {

> 		npctalk "Guided by the light, the Prontera Priests Guild is recruiting new guild members", "", bc_self;

Reviewer 2: ```suggestion
		npctalk "Guided by the light, the Prontera Priest Guild is recruiting new guild members", "", bc_self;
```
